src/workerd/api/node/zlib-util.c++

```diff
@@ -450,4 +547,67 @@ void ZlibUtil::ZlibStream::reset(jsg::Lock& js) {
   }
 }
 
+kj::Array<kj::byte> syncProcessBuffer(ZlibContext& ctx, GrowableBuffer& result) {
+  KJ_DEFER(ctx.close());
+
+  do {
+    // TODO: We respect the chunk size to match node, but it's a bit silly.. should we just
+    // override it for perf?
+    result.addChunk();
+    ctx.setOutputBuffer(kj::ArrayPtr(result.end(), result.available()));
+
+    ctx.work();
+
+    KJ_IF_SOME(error, ctx.getError()) {
+      JSG_FAIL_REQUIRE(Error, error.message);
+    }
+
+    result.adjustUnused(ctx.getAvailOut());
+  } while (ctx.getAvailOut() == 0);
+
+  KJ_IF_SOME(error, ctx.getError()) {
+    JSG_FAIL_REQUIRE(Error, error.message);
+  }
+
+  return result.releaseAsArray();
+}
+
+kj::Array<kj::byte> ZlibUtil::zlibSync(InputSource data, Options opts, ZlibModeValue mode) {
+  ZlibContext ctx;
```

Comment:
You don't seem to call the constructor?

```suggestion
  ZlibContext ctx{};
```

Reply:
In C++, the code I've written invokes the constructor with no arguments.